pkg/insights/metrics.go

```diff
@@ -0,0 +1,37 @@
+package insights
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	insightsMetricsRegistry *prometheus.Registry
+)
+
+func init() {
+	insightsMetricsRegistry = prometheus.NewRegistry()
+	MustRegisterMetricCollectors(
+		collectors.NewGoCollector(),
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+	)
+	startMetricsServer()
```

Comment:
Is it a good practice to start http server within `init` function? I am not sure....I was reading https://go.dev/doc/effective_go#init, but still....

Reply:
I was wondering what might be a good place for it, and the `init` function kind of works because it's executed before any regular code, which means the metrics will be available regardless of anything else breaking, but at the same time, I think you're right and it should be something in `start.go` perhaps, but I'm not too sure where exactly to make sure the metrics are always available. The thing is, IO has multiple "commands" (executables), so I don't know if only the `insights-operator` one should start the metrics server.

Reply:
I think we could add it to https://github.com/openshift/insights-operator/blob/master/pkg/controller/operator.go#L46 method.